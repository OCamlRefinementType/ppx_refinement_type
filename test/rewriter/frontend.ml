open Stdlib

let[@rty] foo =
  let x = (true : int) in
  (v > 3 : int)

let foo x = x + 2;;

foo 1
