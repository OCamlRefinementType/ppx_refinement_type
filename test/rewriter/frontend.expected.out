Type judgement [foo]
|-
fun x -> x + 2
: x:{v:int | true} -> {v:int | v > 3}
